packages/stats/gbstats/gbstats.py

```diff
@@ -319,14 +323,28 @@ def analyze_row(s: pd.Series) -> pd.Series:
 # Convert final experiment results to a structure that can be easily
 # serialized and used to display results in the GrowthBook front-end
 def format_results(
-    df: pd.DataFrame, baseline_index: int = 0
+    df: pd.DataFrame, df_weights: pd.DataFrame, baseline_index: int = 0
 ) -> List[DimensionResponse]:
     num_variations = df.at[0, "variations"]
     results: List[DimensionResponse] = []
+    includes_bandit_weights = df_weights.shape[0] > 0
+    if includes_bandit_weights:
+        df = df.merge(df_weights, on="dimension")
```

Comment:
I'd really love for us to go for an approach that doesn't rely on a merge. This is definitely risky if we change stuff in the future. I think we should consider just calling `get_bandit_weights` inside of `analyze_metric_df` and adding some fields to the returned dataframe there. That way we get rid of:
* calling `preprocess_analysis` twice in both process_analysis and get_bandit_weights
* the unnecessary `var_id_map` and `analysis` arguments to the get_bandit_weights method
* this merge which could break if we change any of the key variables

Reply:
good call, i removed the merge

Reply:
Thanks for simplifying this. I think we can go one more step. Left another comment, so resolving this one.